### PR TITLE
Add stop command

### PR DESCRIPTION
**Pull request by SmajloSlovakian**
i just modified https://github.com/henkelmax/audio-player/pull/57 so it works as the original pull request was inactive

### CODE REVIEW

**Comment by ShiftySpace** [comment]
Dang @SmajloSlovakian  thanks a lot! I am still pretty new to coding and also haven't had the time to figure out how I'd need to modify it for the newer versions. Did me a huge favor :D

**Comment by SmajloSlovakian** [comment]
i'm very new as well... sometimes you just need a clear mind to look at it

File: src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java [hunk 1]
```diff
@@ -41,4 +41,19 @@ public void play(CommandContext<CommandSourceStack> context, @Name("sound") UUID
...
+    @Command("stop")
+    private static int stop(CommandContext<CommandSourceStack> context, @Name("sound") UUID sound) {
+        UUID channelID = PlayerManager.instance().findChannelID(sound);
+
+        if (channelID != null) {
+            PlayerManager.instance().stop(channelID);
+                context.getSource().sendSuccess(() -> Component.literal("Successfully stopped %s.".formatted(sound)), false);
+                return 1;
+        } else {
+            context.getSource().sendFailure(Component.literal("Failed to stop, Could not find %s".formatted(sound)));
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java:54]: `Failed to stop, could not find sound with ID %s`

File: src/main/java/de/maxhenkel/audioplayer/PlayerManager.java [hunk 2]
```diff
@@ -197,7 +197,16 @@ private interface Stoppable {
     }
 
     private record PlayerReference(Stoppable onStop,
-                                   AtomicReference<de.maxhenkel.voicechat.api.audiochannel.AudioPlayer> player) {
+                                   AtomicReference<de.maxhenkel.voicechat.api.audiochannel.AudioPlayer> player, UUID sound) {
+    }
+    @Nullable
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/audioplayer/PlayerManager.java:202]: There should be an empty line

File: src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java [hunk 3]
```diff
@@ -41,4 +41,19 @@ public void play(CommandContext<CommandSourceStack> context, @Name("sound") UUID
         context.getSource().sendSuccess(() -> Component.literal("Successfully played %s".formatted(sound)), false);
     }
 
+    @RequiresPermission("audioplayer.play_command")
+    @Command("stop")
+    private static int stop(CommandContext<CommandSourceStack> context, @Name("sound") UUID sound) {
```
Comment (henkelmax) [inline src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java:46]: The command currently just stops ALL sounds with that ID. This does not seem to be intentional. It should only stop sounds that were played with the play command. I'd suggest a boolean field marking the sound as a sound that was played with the command and filtering by that.

**Review by henkelmax — CHANGES_REQUESTED** [review]

**Comment by SmajloSlovakian** [comment]
oh wow how do i approve everything... i clicked on view reviewed changes and checked that i viewed the files and clicked on review changes but i can't approve? do i really have to just make the same changes you proposed and commit them to my fork?

**Comment by SmajloSlovakian** [comment]
ohh i see how this works... i have to do that because you didn't like edit my code and submit it to me to approve it or somehting... ok... will look at it tomorrow

**Review by henkelmax — CHANGES_REQUESTED** [review]

File: src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java [hunk 4]
```diff
@@ -36,22 +36,23 @@ public void play(CommandContext<CommandSourceStack> context, @Name("sound") UUID
...
-        UUID channelID = PlayerManager.instance().findChannelID(sound);
+        UUID channelID = PlayerManager.instance().findChannelID(sound,true);
 
         if (channelID != null) {
             PlayerManager.instance().stop(channelID);
                 context.getSource().sendSuccess(() -> Component.literal("Successfully stopped %s.".formatted(sound)), false);
                 return 1;
         } else {
-            context.getSource().sendFailure(Component.literal("Failed to stop, Could not find %s".formatted(sound)));
+            context.getSource().sendFailure(Component.literal("Failed to stop, could not find %s".formatted(sound)));
```
[the remark this answers is not in the document]
Comment (SmajloSlovakian) [reply src/main/java/de/maxhenkel/audioplayer/command/PlayCommands.java]: ohh.. sorry, i missed the ID part... i just thought that you meant to change the case of the c in "could"

fixed now

**Pull request merged**